lib/Optimizer/Transforms/DependencyAnalysis.cpp

```diff
@@ -915,6 +1135,27 @@ class DependencyGraph {
                  current + next->numTicks() + successor->numTicks());
   }
 
+  /// Replaces the leaf for \p old_qid (if \p old_qid is part of the graph) with
+  /// \p new_leaf which has \p new_qid by removing the old leaf for \p old_qid
+  /// from the graph metadata and replacing the dependency on the old leaf with
+  /// \p new_leaf.
+  ///
+  /// If \p old_qid was not part of the graph, this has the effect of adding \p
+  /// new_leaf to the graph.
+  ///
+  /// Cleaning up the old leaf is the responsibility of the caller.
+  // TODO: this, replaceRoot, and replaceLeafAndRoot are all a mess.
```

Comment:
> this, replaceRoot, and replaceLeafAndRoot are all a mess.

🤣

Reply:
Fixed. I made the comment focus on the specific issues with it.